### DISCUSSION TITLE
Retrieving the Authorization Code and Obtaining an Access Token and Refresh Token

### DISCUSSION THREAD

**Discussion by liquidmoon09** [spotify-web-api-java/spotify-web-api-java#401] [Q&A] [answered]
After much difficulty and confusion as to how everything works, I've managed to obtain a redirect URI which when clicked on provides me with an authorization code in the new URI its self. 
**Here's my question:** How do I retrieve the authorization code programmatically? It's in the new URI, but then I have to copy and paste it into the code manually and re-run the program to be able to use it just once.

**Comment by dargmuesli** [comment, marked as answer]
The website you redirect to could show the code and maybe a button to copy it or, if you want to get more professional, the website could open your app on a button click using a custom protocol inserting your code automatically.

Here is my example of a website (in Vue) which displays the code and a button to copy it: https://github.com/dargmuesli/jonas-thelemann/blob/eda0385d2b3359f6630ccacea1096552337544a6/src/pages/api/pages/oauth.vue

**Reply by dargmuesli** [reply]
Alternatively, depending on the app you're building, you could add an in-app browsers as well which shows the authorization and is under your full control, but people might not trust your app to not monitor login credentials.
